src/spider/storage/mysql/MySqlStorage.cpp

```diff
@@ -1224,7 +1271,12 @@ auto MySqlMetadataStorage::get_task(StorageConnection& conn, boost::uuids::uuid
             };
         }
         res->next();
-        *task = fetch_full_task(static_cast<MySqlConnection&>(conn), res);
+        auto fetch_task_result = fetch_full_task(static_cast<MySqlConnection&>(conn), res);
+        if (fetch_task_result.has_error()) {
+            static_cast<MySqlConnection&>(conn)->rollback();
+            return StorageErr{fetch_task_result.error(), "Failed to fetch full task"};
+        }
+        *task = fetch_task_result.assume_value();
```

Comment:
```suggestion
        *task = std::move(fetch_task_result.value());
```
Use the move constructor.